### DISCUSSION TITLE
Why would  context be NULL when the response is written asynchrnously

### DISCUSSION THREAD

**Discussion by luxapana** [uNetworking/uWebSockets#1745] [Q&A] [answered]
Hello
Our app replies to the http request asynchronously. The send response code is invoked by calling post of uWS from an application thread. The app crashes as below. This usually occurs after I run a load test, stop the load, and restart the load with new set of connections. Checked the uWS::HttpResponse object and it seem to be valid. If some pre-checks need to be done before trying to send the response what are those?

![image](https://github.com/uNetworking/uWebSockets/assets/45680862/80cabb54-9a93-4e88-aaab-e0212a350a86)
![image](https://github.com/uNetworking/uWebSockets/assets/45680862/3d469616-682b-4a90-aaf4-abeda8e6a70d)

Thanks
M

**Comment by uNetworkingAB** [comment, marked as answer]
The library is single-threaded. If you want to do anything from another thread you must go via Loop::defer which is like io_context->post in boost

**Comment by luxapana** [comment]
I am calling defer to send the response.

**Comment by uNetworkingAB** [comment]
Sounds like you have other issues then. You need to start using AddressSanitizer and ThreadSanitizer to get down to the problem because this report doesn't provide anything of use for me. Build WITH_ASAN=1 and fix your problems

**Reply by luxapana** [reply]
thanks for the nice attitude. I hope the attitude is different for commercially subscribed customers. 

FYI, there aren't any threading and memory issues. The app is designed to be single threaded everywhere (just like uWS) with explicit queues and thread comm mechanism when inter-thread comm is needed. The client creating the load (wrk) creates 40 connections but onAbort is only called for 38. Perhaps there is some **useful info** for you there.

**Reply by uNetworkingAB** [reply]
Oh there is no attitude, I'm just stating objective facts. I cannot do anything with a single image of a single line of code where I know nothing at all about your app. It is technically impossible for me to do anything with such a report, as it lacks any information to go by and has no clear motivation for an issue.

This is not attitude, it's just me presenting the objective facts of the situation.

> FYI, there aren't any threading and memory issues.

and

> The app crashes as below.

are two logically contradicting statements. Clearly you have memory issues and clearly said issues could be diagnosed using AddressSanitizer.

**Reply by luxapana** [reply]
as you can see I have created a discussion, not opened an issue in the hope of getting some useful information on uWS to help diagnose the issue. You seem to be taking it too personally. For example, if you can let me know at which time does uWS set the context to null, it will help me debug this issue. Good to know that this is the nature of the support given, perhaps need to think about moving out to boost or maybe write our own. Thanks for the time.

**Reply by uNetworkingAB** [reply]
I'm glad you enjoyed it
